### ISSUE TITLE
Yet another \item indentation error

### ISSUE THREAD

**Issue by hamdav** [lervag/vimtex#2382]
### Description

I've tried checking out the other \item indentation error issues but they are either claimed to be resolved or not quite the same as this one, so I'm making a new one.
The problem is this: I have a couple of lines hastily written down like so:

    the first item
    the second item
    and so on

And then when I later go to clean up what I've written, I end up with

    \begin{itemize}
        \item the first item
            \item the second item
                 \item the third item
    \end{itemize}

What I am doing is first putting the environment around the lines that will become the list, and then insert \item to the front of the line. If I from the beginning write the items in order then the indentation becomes correct. `=ie` does fix it but it's annoying to have to do that all the time.

### Steps to reproduce

Minimal init.vim:

    all plug#begin('~/.config/nvim/plugged')
    Plug 'lervag/vimtex'
    call plug#end()

1. Open a tex document
2. write two lines
3. surround in `\begin{itemize}` / `\end{itemize}`
4. Insert `\item` at the beginning of the two lines
The first one is correctly indented, the second is indented too much

### Expected behavior

All \items being indented one tab / four spaces

### Actual behavior

Indent level increasing with each \item

### Do you use a latexmkrc file?

no

### VimtexInfo

```yaml
System info:
  OS: macOS 11.2.3 (20D91)
  Vim version: NVIM v0.6.1
  Has clientserver: true
  Servername: /var/folders/mn/7wv53fcj6m95stzmb3nq73bw0000gn/T/nvimSyp5Hp/0

VimTeX project: main
  base: main.tex
  root: /Users/davidhambraeus/Hive/Courses/QIP_II/Excercises/ps8_paper_reading
  tex: /Users/davidhambraeus/Hive/Courses/QIP_II/Excercises/ps8_paper_reading/main.tex
  main parser: current file verified
  document class: article
  packages: aligned-overset amsbsy amsfonts amsgen amsmath amsopn amssymb amstext amsthm a
  source files:
    main.tex
    packages.tex
    mathCommands.tex
    textSettings.tex
  compiler: latexmk
    engine: -pdf
    options:
      -verbose
      -file-line-error
      -synctex=1
      -interaction=nonstopmode
    callback: 1
    continuous: 1
    executable: latexmk
  viewer: General
  qf method: LaTeX logfile
```

**Comment by lervag** [comment]
I believe the problem is that the specified workflow sort of breaks the expectations used for auto indentation here. That is, given:

```tex
\begin{itemize}
  \item first
second
\end{itemize}
```

When you start adding the `\item` in front, e.g. with `I\ite`, you get this (`|` marks the cursor):

```tex
\begin{itemize}
  \item first
\ite|second
\end{itemize}
```

Now continue in insert mode by adding "m", you get:

```tex
\begin{itemize}
  \item first
    \itemsecond
\end{itemize}
```

This is because the indentation is triggered by the `\item`, which does not look _forward_ in the line. But the indentation rule does not recognize `\itemsecond` as an actual bullet and the lien is therefore indented as if it is not a new separate `\item` line.

As a "proof" of this, you can try the same story as above, except add the initial space. That is, do `0i <left>\item`, and the indentation is now as expected:

```tex
\begin{itemize}
  \item first
\ite| second
  % continue with "m" →
  \item second
\end{itemize}
```

I've tried to thing of a "solution" to this, but I'm afraid I don't immediately see any. Or, one possibility is to relax the regex used for the indentation:

https://github.com/lervag/vimtex/blob/1eeb881a879528cd361fe8c2efd9b5d1189ecedb/indent/tex.vim#L233

If we remove `\>` at the above line, then things work. But this _could_ break indentation if you for some reason wrote something like:

```tex
\begin{env}
  Some text where you for some reason break the line here
  \itemcmd and continue here with some random command starting with
    "\item..". The indentation is now broken.
\end{env}
```

I might consider going for this, as the broken scenario seems very unlikely. But still, we will have to choose between things that are not perfect.

I would be happy to hear more opinions on this matter. Or better; suggestions for other ways to handle this.

**Comment by hamdav** [comment]
Would it be possible to trigger the indentation on `\item ` (with the space) instead of just `\item`?

**Comment by lervag** [comment]
Yes, that is possible, but it would ruin auto indent on `\item[`. However, that may be OK, as I could just add that as an additional indent key...

**Comment by lervag** [comment]
I believe this is actually a superior behaviour. It also means we can remove some code we use to fix an indent issue on completion (#2179, #2156). I'm pushing the updates now.
